### PR TITLE
update .devcontainer/README.md

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Loong Dai <loong.dai@intel.com>

Commit Message:
Additional Description:
Risk Level: Low
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .devcontainer/README.md [hunk 1]
```diff
@@ -15,6 +15,12 @@ This task is needed to run everytime after:
 
 There are additional tools for VS Code located in [`tools/vscode`](../tools/vscode) directory.
 
+Note:
+Current config assums you use `vscode` to develop. If you use other user like `root` to develop, please update `remoteUser` field in [`devcontainer.json`](devcontainer.json) as below:
+```json
+"remoteUser": "root",
```
Comment (phlax) [inline .devcontainer/README.md:21]: im not sure we want to encourage people to dev as root

Comment (lizan) [reply .devcontainer/README.md:21]: yeah don't encourage this. The `vscode` user has sudo permission anyway.

Comment (daixiang0) [reply .devcontainer/README.md:21]: In my dev environment, I setup Envoy repo via `root` user, then `vscode` failed to create link or inherit env values(something like `sudo -E`). I think it is a solution for users who do development via `root` user in an exclusive virtual/physical machine.

Comment (daixiang0) [reply .devcontainer/README.md:21]: Error like below:
```
[3909 ms] Start: Run in container: mkdir -p '/build/.vscode-server/bin' && ln -snf '/vscode/vscode-server/bin/linux-x64/dfd34e8260c270da74b5c2d86d61aee4b6d56977' '/build/.vscode-server/bin/dfd34e8260c270da74b5c2d86d61aee4b6d56977'
[3933 ms] 
[3934 ms] ln: failed to create symbolic link '/build/.vscode-server/bin/dfd34e8260c270da74b5c2d86d61aee4b6d56977': Permission denied
[3935 ms] Exit code 1
```

**Comment by daixiang0** [comment]
Also I do not have a user called `vscode`:
```
$ id vscode
id: ‘vscode’: no such user
```

**Comment by daixiang0** [comment]
It is bad news for users who develop as `root`. I will close it.

File: .devcontainer/README.md (hunk 1 above)
Comment (jason9015) [reply .devcontainer/README.md:21]: > daixiang0
解决了吗？老哥

**Pull request closed without merging**